core/src/main/java/com/novoda/merlin/HttpRequestMaker.java

```diff
@@ -11,8 +11,9 @@
 
     @Override
     public Request head(Endpoint endpoint) {
+        HttpURLConnection urlConnection = null;
```

Comment:
can you `try with resources`?

http://tutorials.jenkov.com/java-exception-handling/try-with-resources.html

Reply:
API level is too low on this project and has to be `AutoCloseable` 😢

Reply:
It would work only if `HttpUrlConnection` extends from `AutoCloseable`